src/hooks/useAppState/middleware.js

```diff
@@ -16,34 +18,44 @@ const applyMiddleware = (dispatch, getState) => (action) => {
   switch (type) {
     case types.DELETE_ITEM_BY_KEY: {
       const state = getState();
-      const itemToRemove = state.items.find((item) => item.key === data.key);
+      const itemToRemove = selectItems(state).find((item) => item.key === data.key);
       if (itemToRemove) {
-        deleteItem(state.listId, itemToRemove.id);
+        deleteItem(selectListId(state), itemToRemove.id);
       }
       break;
     }
 
     case types.UPDATE_ITEM_BY_KEY: {
       const state = getState();
-      const itemToUpdate = state.items.find((item) => item.key === data.key);
+      const itemToUpdate = selectItems(state).find((item) => item.key === data.key);
       if (itemToUpdate) {
-        updateItem(state.listId, itemToUpdate.id, data.text);
+        updateItem(selectListId(state), itemToUpdate.id, data.text);
```

Comment:
Same here with the listID, it's a bit more complicated now so moved it to a helper function